update-gyp.py

```diff
@@ -33,7 +33,25 @@
 
         print("Unzipping...")
         with tarfile.open(tar_file, "r:gz") as tar_ref:
-            tar_ref.extractall(unzip_target)
+            def is_within_directory(directory, target):
+
+                abs_directory = os.path.abspath(directory)
+                abs_target = os.path.abspath(target)
+
+                prefix = os.path.commonprefix([abs_directory, abs_target])
+
+                return prefix == abs_directory
+
+            def safe_extract(tar, path=".", members=None, *, numeric_owner=False):
+
+                for member in tar.getmembers():
+                    member_path = os.path.join(path, member.name)
+                    if not is_within_directory(path, member_path):
+                        raise Exception("Attempted Path Traversal in Tar File")
+
+                tar.extractall(path, members, numeric_owner)
```

Comment:
https://docs.python.org/3/library/tarfile.html#tarfile.TarFile.extract
This ___must___ be `tar.extractall(path, members, numeric_owner=numeric_owner)`